go/teams/chain.go

```diff
@@ -1447,13 +1468,30 @@ func (t *TeamSigChainPlayer) addInnerLink(
 		return res, NewTeamDeletedError()
 	case libkb.LinkTypeDeleteUpPointer:
 		return res, NewTeamDeletedError()
-	case libkb.LinkTypeLegacyTLFUpgrade:
-		// This link type is not really understood but is processed as a no-op for forward compatibility.
-		// When implementing this for real (or deleting it) be sure to:
-		// - Bust the TeamData cache
-		// - Update SigchainV2Type.RequiresAdminPermission
-		// - Add checks here that the signer is an admin and that this is an implicit team. If those are intended.
+	case libkb.LinkTypeKBFSSettings:
+		err = libkb.PickFirstError(
+			hasKBFSSettings(true),
+			allowInflate(false),
+			hasPrevState(true),
+			hasName(false),
+			hasMembers(false),
+			hasParent(false),
+			hasSubteam(false),
+			hasInvites(false),
+			hasPerTeamKey(false),
+			hasCompletedInvites(false))
```

Comment:
Add `hasSettings(false)`. Also add `hasKBFSSettings(false)` to all other link cases.

These `has*` are pretty error prone. An idea to make it better is below. No pressure to do that this PR though.
```
type Tristate int
Tristate_DISALLOW = 0
Tristate_REQUIRE = 1
Tristate_OPTIONAL = 2

type LinkRequirements struct {
  allowInflate bool
  prevState bool
  name Tristate
  members Tristate
  parent Tristate
...
}
```

Reply:
decided not to fix it here

Reply:
Wait.. that's fine not to do the new thing but you gotta do this I think
> Add hasSettings(false). Also add hasKBFSSettings(false) to all other link cases.

Reply:
i did add `hasKBFSSettings` everywhere.  But not `hasSettings`.